sdk/core/Azure.Core.TestFramework/src/ClientTestBase.cs

```diff
@@ -21,13 +25,82 @@ public abstract class ClientTestBase
         private static readonly IInterceptor s_diagnosticScopeValidatingInterceptor = new DiagnosticScopeValidatingInterceptor();
         private static Dictionary<Type, Exception> s_clientValidation = new Dictionary<Type, Exception>();
         public bool IsAsync { get; }
-
+        protected static TestLogger Logger { get; set; }
         public bool TestDiagnostics { get; set; } = true;
+        public RecordedTestMode Mode { get; set; }
```

Comment:
ClientTestBase has nothing to do with recordings. RecordedTestMode shouldn't be defined here.

Reply:
We need to get clear then on what we want to do here. 

The current thought is we will have class OneTimeSetupRecordBase.cs (currently wrapped in ManagementRecordTestBase.cs) for one time setup with ARM template or programmatically and then the current way things work (currently RecordedTestBase.cs). The shared functionality would be in ClientTestBase.cs which they both inherit from. 

I think you are still of the view that we merge OneTimeSetupRecordBase (currentyl ManagementRecordTestBase) into RecordedTestBase.cs? If we do that, then yeah we don't need any changes to ClientTestBase at all.

I think the majority view is that we keep these separated. That is, we have an explicitly class for onetimesetup and another for doing things the current way.

Reply:
To me, the OneTimeSetup recording is just another feature of recorded tests. I don't see a reason to have it in a separate file.